### DISCUSSION TITLE
Keybindings configuration

### DISCUSSION THREAD

**Discussion by kyp0717** [nix-community/kickstart-nix.nvim#51] [Q&A] [answered]
Hello.
I was wondering if there was a standard or best-practice approach for keybindings.
I know there is a file called `keymaps.lua` and there are also keybindings in `telescopes.lua` and in other places.  It is difficult to track down all the keybindings. I try the `map` and `Telescope keymap` command but it returns to many items.

I am looking to see if there is a keymap for `neotree` but can't seem to find it.
If a keymap for `neotree` does not exist, what is the recommended approach to create keymap for this plugin?

Please advise.  Thanks.

**Comment by mrcjkb** [comment]
Hey :wave: 

There's no "recommended way" to define keymaps.
I tend to put plugin-specific keymaps in the respective script that I use to configure the plugin, because if I remove the plugin I also remove all the keymaps for it.

This template does not include neotree, which is why you can't find any keymaps for it.

**Reply by kyp0717** [reply]
Thank you for your respond.  It was very quick and very helpful.  Should I create PR and add neotree keymap for this repo?

**Reply by mrcjkb** [reply]
> Should I create PR and add neotree keymap for this repo?

No, I do not see a need to add neotree to this template.
I'm personally [not a fan of project-drawer style plugins](http://vimcasts.org/blog/2013/01/oil-and-vinegar-split-windows-and-project-drawer/).

**Comment by kyp0717** [comment]
To follow up on `neotree`, how is it that `neotree` is enabled in kickstart nvim when I can't seem to find the lua configuration file that enable this plugin?  In other words, where is the statement `require("neotree")` in this configuration.  
Please help me understand how configuration operate under kickstart nvim because I know I may be missing something in my knowledge.

**Reply by mrcjkb** [reply, marked as answer]
This template is not kickstart.nvim. It follows its own philosophy. Regarding neotree, please see [my other comment](https://github.com/nix-community/kickstart-nix.nvim/discussions/51#discussioncomment-11374225).

**Comment by kyp0717** [comment]
Thank you for suggesting a new way to navigate files in vim.  I will try this new method!  It was very insightful.

**Reply by mrcjkb** [reply]
it's not exactly new 😄 

By the way, my preferred file browser plugin for Neovim is [oil.nvim](https://github.com/stevearc/oil.nvim) (combined with FzfLua or telescope.nvim).

**Reply by kyp0717** [reply]
I just saw a demo on oil.  Looks amazing.  I will need add this as a plugin to kickstart I'm guessing.  Is this correct?

**Reply by mrcjkb** [reply]
yes, it isn't part of this template yet. 
See

 - https://github.com/nix-community/kickstart-nix.nvim/issues/43

I'm not 100 % sure if I want to add it, but so far I haven't encountered anyone who didn't like oil.nvim.

**Reply by kyp0717** [reply]
Just added oil as plugin.  OMG...what a great plugin!
